### PR TITLE
fix: personal spaces can be made private without a pro plan

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

Fixes ENG-140
Personal spaces can always be made private regardless of pro plan
Once affected spaces are private again, we don't allow to open them to avoid confusion

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [ ] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by BlankParticle** [comment]
* **#776** <a href="https://app.graphite.dev/github/pr/un/inbox/776?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BlankParticle and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by McPizza0** [comment]
@BlankParticle i think we should change this to never allow personal spaces to be public in the first place
will be less confusing for people

**Comment by BlankParticle** [comment]
> @BlankParticle i think we should change this to never allow personal spaces to be public in the first place will be less confusing for people

there might be use cases for people to have open personal spaces, also we need to alllow people who have accidently made their spaces open

**Comment by McPizza0** [comment]
> there might be use cases for people to have open personal spaces, also we need to alllow people who have accidently made their spaces open

maybe only a tiny few usecases
most of the time they can just add people to the space
If someone accidentally made it open; we can account for that.

if personal space is true, allow it to be set to private, but dont allow to be set to open (open is disabled)

**Comment by BlankParticle** [comment]
@McPizza0, has done as you asked, also added the missing backend checks

File: apps/web/src/app/[orgShortcode]/[spaceShortcode]/settings/page.tsx [hunk 1]
```diff
@@ -166,15 +166,20 @@ export default function SettingsPage() {
...
-              showSaved={setShowSaved}
-              isSpaceAdmin={isSpaceAdmin}
-            />
-            <VisibilityField
-              initialValue={spaceSettings?.settings?.type}
+              initialValue={spaceSettings.settings.color}
               showSaved={setShowSaved}
               isSpaceAdmin={isSpaceAdmin}
             />
+            {/* Show visibility field only if the space is not a personal space or if it is a personal space and it is open so that users can change the visibility */}
```
Comment (McPizza0) [inline apps/web/src/app/[orgShortcode]/[spaceShortcode]/settings/page.tsx:173]: should always show the field
otherwise ux is confusing

**Review by McPizza0 — CHANGES_REQUESTED** [review]

File: apps/web/src/app/[orgShortcode]/[spaceShortcode]/settings/page.tsx (hunk 1 above)
Comment (BlankParticle) [reply apps/web/src/app/[orgShortcode]/[spaceShortcode]/settings/page.tsx:173]: so just disable the open option on personal spaces?

Comment (McPizza0) [reply apps/web/src/app/[orgShortcode]/[spaceShortcode]/settings/page.tsx]: yes

Comment (BlankParticle) [reply apps/web/src/app/[orgShortcode]/[spaceShortcode]/settings/page.tsx]: ok then the PR is already updated and ready

**Review by McPizza0 — APPROVED** [review]

**Pull request merged**